Tools scala3 zio r17

pmeheut opened the pull request:
Attempted solution to: https://github.com/ghostdogpr/caliban/issues/1239

Used https://github.com/scalameta/scalafmt/issues/3035#issuecomment-1007187349 suggestion to load scalafmt dynamically, allowing to generate a Scala 2.12/2.13 & 3 of caliban-tools and using the sbt-codegen plugin in Scala 3.

pmeheut committed:
Now working with RC21

pmeheut committed:
Now working with RC21

pmeheut committed:
Working in Scala 3 without scalafmt

pmeheut committed:
Working in Scala 3 without scalafmt

pmeheut committed:
Now using scalafmt-interfaces, no more direct dependencies

pmeheut committed:
Now using scalafmt-interfaces, no more direct dependencies

pmeheut committed:
Merge branch 'tools_scala3' of github.com:pmeheut/caliban into tools_scala3

pmeheut committed:
Now using BuildInfo to load the correct version of scalafmt in Formatter

pmeheut committed:
Cleaned a few imports

guizmaii commented on build.sbt:
```diff
@@ -1,48 +1,49 @@
+val zioConfigVersion = "1.0.10"
+val zqueryVersion = "0.2.10"
+val zioJsonVersion = "0.1.5"
+val zioHttpVersion = "1.0.0.0-RC17"
```
@pmeheut Please use the project formatter

pmeheut replied:
Done.

guizmaii replied:
Thanks

pmeheut committed:
Fixed foromatting

pmeheut committed:
Fixed formatting for build.sbt too

ghostdogpr reviewed (approved):
Looks good, just a tiny comment!

pmeheut committed:
removed a commented line in build.sbt

guizmaii commented:
@ghostdogpr @pmeheut Shouldn't we write test to validate that it's working and continue to work over time?

Maybe a new project using Scala3 in the scripted test of the codegen module?

pmeheut commented:
I can do this if you are in no hurry. Current tests in codegen are Scala 2.12 only I think. Should I make them cross-version or do something specific for Scala 3?

guizmaii commented:
@pmeheut The best would be to create a third test project, not modifying the two others because they are testing their own stuff.

Sadly, I think we'll be limited by sbt. I remember being forced to use Scala 2.12 instead of 2.13 in the test project because of sbt which is using Scala 2.12 but maybe I was wrong. Maybe I could have used Scala 2.13 which means that you maybe can write this third test project using Scala3.

Can you please give it a try, maybe?

pmeheut commented:
I'll give it a try and let you know.

scala-steward committed:
Update laminext:core, laminext:fetch, ... to 0.14.3 (#1249)

nicoburniske committed:
added case in BigInt & BigDecimal default ScalarDecoder to account for Strings (#1250)

scala-steward committed:
Update play, play-akka-http-server to 2.8.13 (#1251)

cornerman committed:
use fully-qualified name for Vector in generated code (#1254)

This is to avoid name-clashes with types from the graphql schema, i.e.,
if a type is called Vector.

Example:
enum ImageType {
  Raster
  Vector
}

scala-steward committed:
Update scala-library to 2.13.8 (#1257)

ghostdogpr committed:
Fix typo

ghostdogpr committed:
[ci skip] update docs

ghostdogpr committed:
Update config.yml (#1259)

frekw committed:
feat: Relay cursors/connections (#1196)

ghostdogpr committed:
[ci skip] update docs

ghostdogpr committed:
Resolve directive argument variables (#1263)

smiklos committed:
support uuid as arg encoder (#1266)

* support uuid as arg encoder

* test the uuid encoder

* reuse string value for uuid encoder

iRevive committed:
Configure mdoc and fix documentation (#1268)

ghostdogpr committed:
[ci skip] update docs

scala-steward committed:
Update http4s-blaze-server to 0.23.8 (#1273)

iRevive committed:
docs module: do not enforce a specific scala version (#1274)

Fluxx committed:
Render directives at the top of the schema  (#1270)

* Render directives at the top of the schema

* Remove extranous comma

* Remove repeatable statement from directives

scala-steward committed:
Update sbt-mdoc to 2.3.0 (#1276)

Fluxx committed:
Add missing directive rendering for derrived types (#1275)

scala-steward committed:
Update cats-effect to 3.3.5 (#1277)

ghostdogpr committed:
Update version in docs

ghostdogpr committed:
Fix typo

ghostdogpr committed:
[ci skip] update docs

ghostdogpr committed:
Fix typos

ghostdogpr committed:
Merge branch 'master' of https://github.com/ghostdogpr/caliban

ghostdogpr committed:
Retrigger docs

ghostdogpr committed:
[ci skip] update docs

scala-steward committed:
Update http4s-blaze-server to 0.23.9 (#1278)

ghostdogpr committed:
Flatten SelectionBuilder zipping (#1212)

* Flatten SelectionBuilder zipping

* Fix ClientWriterViewSpec

* Add attribution for Zippable

ghostdogpr committed:
Improve docs

ghostdogpr committed:
[ci skip] update docs

iRevive committed:
Add contextual interop between cats-effect and ZIO (#1246)

jyoo980 committed:
Adding parser logic for inputs and enums with empty bodies (#1283)

scala-steward committed:
Update sbt to 1.6.2 (#1284)

scala-steward committed:
Update scala3-library, ... to 3.1.1 (#1285)

* Update scala3-library, ... to 3.1.1

* Update config.yml

Co-authored-by: Pierre Ricadat <ghostdogpr@users.noreply.github.com>

scala-steward committed:
Update http4s-blaze-server to 0.23.10 (#1286)

frekw committed:
fix: flaky Relay test (#1287)

paulpdaniels committed:
Improve adapter ergonomics (#1288)

paulpdaniels committed:
Add schema reporting (#1280)

* Add apollo schema reporting

* Add the schema reporting docs

* Improve reporting interface

* Fix mdoc

ghostdogpr committed:
Fix problems

ghostdogpr committed:
[ci skip] update docs

guizmaii committed:
CompileTime plugin: Add missing links in doc (#1289)

scala-steward committed:
Update zio-interop-cats to 3.2.9.1 (#1290)

scala-steward committed:
Update sbt-scalajs, scalajs-compiler, ... to 1.9.0 (#1292)

scala-steward committed:
Update akka-http to 10.2.8 (#1291)

Fluxx committed:
Support passing interfaces to Types.makeObject (#1294)

scala-steward committed:
Update tapir-akka-http-server, tapir-core, ... to 0.19.4 (#1297)

scala-steward committed:
Update sbt-buildinfo to 0.11.0 (#1300)

scala-steward committed:
Update compilerplugin, scalapb-runtime to 0.11.9 (#1301)

SvenW committed:
Make sure relay validation errors gets propagated to the user (#1302)

scala-steward committed:
Update sbt-mdoc to 2.3.1 (#1303)

mdulac committed:
Handle Long as String (#1304)

dojusa committed:
Fix relay connection from list (#1307)

ghostdogpr committed:
Make sure the subscription stream is interruptible (#1308)

scala-steward committed:
Update cats-effect to 3.3.6 (#1309)

uryyyyyyy committed:
chore: update zio-http (#1216)

* chore: update to tapir 0.20.0

* Fix sttp ws issue

Co-authored-by: kokishibata <koki@anymindgroup.com>
Co-authored-by: Pierre Ricadat <ghostdogpr@gmail.com>

dojusa committed:
Add forward and backward pagination (#1310)

sergeykolbasov committed:
Fix custom directive parsing in Scala 3 impl (#1313)

ghostdogpr committed:
Update version in docs

ghostdogpr committed:
[ci skip] update docs

iRevive committed:
Contextual cats interop. Documentation (#1279)

* Add documentation regarding contextual cats-effect interop

* Mention `Dispatcher` in the effect constraints

ghostdogpr committed:
[ci skip] update docs

scala-steward committed:
Update tapir-akka-http-server, tapir-core, ... to 0.20.1 (#1314)

scala-steward committed:
Update akka-http to 10.2.9 (#1316)

frekw committed:
Add `Field.targets` to expose the type conditions of a field (#1315)

scala-steward committed:
Update cats-effect to 3.3.7 (#1318)

scala-steward committed:
Update compilerplugin, scalapb-runtime to 0.11.10 (#1320)

scala-steward committed:
Update http4s-blaze-server to 0.23.11 (#1323)

scala-steward committed:
Update cats-effect to 3.3.8 (#1324)

frekw committed:
support lower-case content-type header (#1321)

* support lower-case content-type header

* add test

scala-steward committed:
Update sbt-scalajs-crossproject to 1.2.0 (#1327)

scala-steward committed:
Update cats-effect to 3.3.9 (#1328)

scala-steward committed:
Update play, play-akka-http-server to 2.8.14 (#1329)

scala-steward committed:
Update sbt-mdoc to 2.3.2 (#1330)

ghostdogpr committed:
Update README.md

ghostdogpr committed:
Update README.md

scala-steward committed:
Update play, play-akka-http-server to 2.8.15 (#1331)

scala-steward committed:
Update cats-parse to 0.3.7 (#1332)

scala-steward committed:
Update cats-effect to 3.3.10 (#1333)

scala-steward committed:
Update cats-effect to 3.3.11 (#1336)

sergeykolbasov committed:
Fix parsing of empty comment lines in Scala 3 (#1337)

ghostdogpr committed:
Fix formatting

scala-steward committed:
Update zio, zio-streams, zio-test, ... to 1.0.14 (#1340)

moonkev committed:
make websocket stream interruptible in TapirAdapter (#1342)

paulpdaniels committed:
Add federation V2 (#1344)

scala-steward committed:
Update zio-interop-reactivestreams to 1.3.10 (#1346)

anasbarg committed:
Add BusinessChat.io to the list of adopters (#1349)

guizmaii committed:
Add Conduktor to the list of adopters (#1350)

ghostdogpr committed:
Sort adopters alphabetically

kyri-petrou commented:
Hi there 👋 Are there plans to include this in the next release of caliban-tools by any chance?

joheriks committed:
Fix rendering of object arguments to directives (#1353)

scala-steward committed:
Update monix to 3.4.1 (#1357)

ngbinh committed:
adding Anduin Transactions to the adopter list (#1358)

scala-steward committed:
Update zio-interop-reactivestreams to 1.3.12 (#1359)

frekw committed:
Validate input types (#1347)

* wip: validate input types

* wip

* fix test

* fix more tests

* allow null coercion

* fix tests

* revert isINputType

* Harmonize errors

* Renamae to VariablesCoercer

* only add primitive types when doing value coercion

* wip better errors

* wip tests

* fix tests

* add input object test

* remove old test

* remove metals.sbt

* fix imports

pmeheut commented:
> Hi there 👋 Are there plans to include this in the next release of caliban-tools by any chance?

As soon as my real work stops using all my brain, I'll work on it.

strokyl committed:
Update tapir: 0.20.1 -> 0.20.2; zhttp: 1.0.0.0-RC25 -> 1.0.0.0-RC27 (#1361)

paulpdaniels committed:
Fix cache poisoning (#1362)

scala-steward committed:
Update circe-core, circe-generic, ... to 0.14.2 (#1364)

ollyw committed:
Support generated views with more than 22 fields (#1363)

scala-steward committed:
Update cats-effect to 3.3.12 (#1366)

paulpdaniels committed:
Add federation v2 docs (#1348)

ghostdogpr committed:
[ci skip] update docs

ghostdogpr committed:
Update version in docs

ghostdogpr committed:
[ci skip] update docs

chaoky committed:
upgrade zio-config to 2.0.0 (#1367)

scala-steward committed:
Update http4s-blaze-server to 0.23.12 (#1368)

moonkev committed:
Properly validate floating point numbers on input (#1370)

joheriks committed:
Escape quotes in description strings (#1372)

Escape " with \" in normally quoted strings and """ with \""" in
triple-quoted strings.
https://spec.graphql.org/October2021/#sec-String-Value

paulpdaniels committed:
Improve persisted query performance by caching the document (#1371)

* improve persisted query performance by caching the document instead of the unparsed string.

* address comments

joheriks committed:
Escape special characters in normally quoted strings (#1373)

guymers committed:
Correct field types for interface fields on union (#1375)

Ensure that when an interface fragment is used on a union that the
fields its selected have the correct type instead of always being
`String`.

guymers committed:
Coerce non-null values into lists of size one (#1376)

* Coerce non-null values into lists of size one

As per the specification, if a value passed as an input to a list type
is not a list and is not null then it should be coerced into a list of
size one.

* Make work on Scala 3

alterationx10 committed:
Client auto-gen Option conflict (#1377)

* have ClientWriter print scala.Option instead of Option

* Fix replace case where Option may appear multiple times

* Add test for various returns of GQL Option and scala.Option

* run fmt

* update ClientWriterViewSpec

* SchemaWrite should also print scala.Option instead of Option

* add a scripted test with a schema containing type Option

* flatten scripted compile tests

* make helper method for safe Option string replacement

paulpdaniels committed:
add support for graphql-transport-ws (#1354)

* add support for graphql-transport-ws

guymers committed:
Set api mappings for ZIO (#1381)

gerdreiss committed:
Upgrade Scala.js to 1.10.0 (#1383)

* Upgrade Scala.js to 1.10.0

* Upgrade Scala.js to 1.10.0

* use cross version for scalajs-java-securerandom

Co-authored-by: Gerd Reiss <gerdreiss@@users.noreply.github.com>

jeffmay commented:
Is there any plan to release this? I would like to adopt Scala 3 for my Caliban project, but I can't use the code generation tools, which breaks my build.

iida-hayato committed:
update documenation to align with updated ZQuery (#1384)

- https://github.com/zio/zio-query/blob/44c763e17f975e808c81715d50a9ca4c0a77978a/docs/overview/creating_data_sources.md

ghostdogpr committed:
[ci skip] update docs

rolang committed:
update tapir to 1.0.1 (#1386)

* update tapir to 1.0.1

* update examples

scala-steward committed:
Update compilerplugin, scalapb-runtime to 0.11.11 (#1391)

scala-steward committed:
Update play, play-akka-http-server to 2.8.16 (#1392)

scala-steward committed:
Update zio, zio-streams, zio-test, ... to 1.0.15 (#1393)

scala-steward committed:
Update scala-java-time to 2.4.0 (#1399)

scala-steward committed:
Update sbt-scalajs, scalajs-compiler, ... to 1.10.1 (#1400)

scala-steward committed:
Update cats-effect to 3.3.13 (#1403)

scala-steward committed:
Update cats-parse to 0.3.8 (#1404)

scala-steward committed:
Update scala-library to 2.12.16 (#1401)

* Update scala-library to 2.12.16

* Update config.yml

Co-authored-by: Pierre Ricadat <ghostdogpr@users.noreply.github.com>

scala-steward committed:
Update scala3-library, ... to 3.1.3 (#1402)

* Update scala3-library, ... to 3.1.3

* Update config.yml

Co-authored-by: Pierre Ricadat <ghostdogpr@users.noreply.github.com>

pmeheut committed:
Working again on codegen for scala3 and its tests

pmeheut committed:
Working again on codegen for scala3 and its tests

pmeheut committed:
Working again on codegen for scala3 and its tests

pmeheut committed:
Working again on codegen for scala3 and its tests

pmeheut committed:
Scala 3 codegen plugin now working

pmeheut committed:
Scala 3 codegen plugin now working

scala-steward committed:
Update sbt to 1.7.0 (#1406)

scala-steward committed:
Update cats-effect to 3.3.14 (#1409)

scala-steward committed:
Update sbt to 1.7.1 (#1408)

scala-steward committed:
Update zio, zio-streams, zio-test, ... to 1.0.16 (#1411)

pmeheut committed:
build.sbt files are back to normal

pmeheut committed:
build.sbt files are back to normal

pmeheut committed:
Merge branch 'tools_to_merge_1' of github.com:pmeheut/caliban into tools_to_merge_1

pmeheut committed:
Switched to scala 3.1.3 in codegen-sbt/src/sbt-test/compiletime-codegen/test-compile

pmeheut committed:
Tapir to 1.0.2

ghostdogpr commented:
Closing old PRs, but feel free to reopen it targeting the `series/2.x` branch which is the new default.
